compiler/rustc_resolve/src/lib.rs

```diff
@@ -1969,11 +1994,11 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
     }
 
     fn record_use(&mut self, ident: Ident, used_binding: NameBinding<'ra>, used: Used) {
-        self.record_use_inner(ident, used_binding, used, used_binding.warn_ambiguity);
+        self.smart().record_use_inner(ident, used_binding, used, used_binding.warn_ambiguity);
     }
 
-    fn record_use_inner(
-        &mut self,
+    fn record_use_inner<'r>(
+        mut self: SmartResolver<'r, 'ra, 'tcx>,
```

Comment:
```suggestion
        &mut self
```
Never called from import resolution, can use `res_mut` higher on the stack.